go.mod

```diff
@@ -6,6 +6,23 @@ replace github.com/hashicorp/boundary/api => ./api
 
 replace github.com/hashicorp/boundary/sdk => ./sdk
 
+replace (
+	github.com/docker/cli => github.com/docker/cli v20.10.3-0.20221013132413-1d6c6e2367e2+incompatible // 22.06 master branch
+	github.com/docker/docker => github.com/docker/docker v20.10.3-0.20221013203545-33ab36d6b304+incompatible // 22.06 branch
+	github.com/moby/buildkit => github.com/moby/buildkit v0.10.1-0.20220816171719-55ba9d14360a // same as buildx
+
+	github.com/opencontainers/runc => github.com/opencontainers/runc v1.1.2 // Can be removed on next bump of containerd to > 1.6.4
+
+	// For k8s dependencies, we use a replace directive, to prevent them being
+	// upgraded to the version specified in containerd, which is not relevant to the
+	// version needed.
+	// See https://github.com/docker/buildx/pull/948 for details.
+	// https://github.com/docker/buildx/blob/v0.8.1/go.mod#L62-L64
+	k8s.io/api => k8s.io/api v0.22.4
+	k8s.io/apimachinery => k8s.io/apimachinery v0.22.4
+	k8s.io/client-go => k8s.io/client-go v0.22.4
+)
+
```

Comment:
This is needed due to the warning shown here: https://golang.testcontainers.org/quickstart/gotest/

This should be able to be removed in the next version of testcontainers.

Reply:
Wow this is gross 😢. Can we use an older version of testcontainers and avoid this until they release the next version? I think we really really want to avoid this dependency.

Reply:
I'm starting to think this testcontainers thing may not be a suitable dependency at all considering the tree it brings in. There are alternative ways of doing this, for example, https://github.com/ory/dockertest, which has a much smaller dependency closure.

Reply:
I would suggest using dockertest, since we already have it as a dep, `boundary dev` uses it.

Reply:
Whew. Took a bit, but I think I got things working with `dockertest`. Doing some more testing on it.